website/docs/components/code-block/partials/code/how-to-use.md

```diff
@@ -0,0 +1,148 @@
+## How to use this component
+
+The basic invocation requires `@value` to be passed.
+
+```handlebars
+<Hds::CodeBlock
+  @value="aws ec2 --region us-west-1 accept-vpc-peering-connection"
+/>
+```
+
+### Title and description
+
+Optionally, you can pass a title and/or a description.
+
+```handlebars
+<Hds::CodeBlock
+  @language="bash"
+  @value="aws ec2 --region us-west-1 accept-vpc-peering-connection"
+as |CB|>
+  <CB.Title>
+    Acceptance instructions
+  </CB.Title>
+</Hds::CodeBlock>
+```
+
+```handlebars
+<Hds::CodeBlock
+  @language="bash"
+  @value="export VAULT_ADDR=https://0242ac170030.aws.hcp.dev:8200;
+export VAULT_NAMESPACE=admin"
+as |CB|>
+  <CB.Description>
+    Export your cluster’s public URL and the default <a>namespace</a> called admin.
+  </CB.Description>
+</Hds::CodeBlock>
+```
+
+### Language
+
+The `language` argument sets the syntax highlighting used. We only support a the following languaged: `bash`, `go`, `hcl`, `json`, `log`, `shell-session` and `yaml`. If you need additional languages <LinkTo class="doc-link-generic" @route="show" @model="about/support">contact the Design Systems Team</LinkTo>
+
+```handlebars
+<Hds::CodeBlock
+  @language="go"
+  @value="package main
+import fmt
+func main() {
+  fmt.Println(helloWorld)
+}"
+/>
+```
+
+### Copy button
+
+Set `hasCopyButton` to `true` to display a button for users to copy `CodeBlock` content to their computer clipboard.
+
+```handlebars
+<Hds::CodeBlock
+  @language="javascript"
+  @hasCopyButton={{true}}
+  @value="let codeLang=`JavaScript`;
+console.log(`I am ${codeLang} code`);"
+/>
+```
+
+### Line numbers
+
+Line numbers are displayed by default. Set `hasLineNumbers` to `false` to hide them.
+
+```handlebars
+<Hds::CodeBlock
+  @language="javascript"
+  @hasLineNumbers={{false}}
+  @value="let codeLang=`JavaScript`;
+console.log(`I am ${codeLang} code`);"
+/>
+```
+
+### Line wrapping
+
+By default, long lines of code will overflow the `CodeBlock` container requiring users to scroll to view the full content. Setting `hasLineWrapping` to `true` will wrap long lines of code instead.
+
+```handlebars
+<Hds::CodeBlock
+  @language="javascript"
+  @hasLineWrapping={{true}}
+  @value="console.log(`I am JavaScript code`, `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam`);"
+/>
+```
+
+### Highlight lines
+
+Highlight either individual code lines or a range of code lines. (Examples: "2, 4", "6-10")
+
+```handlebars
```

Comment:
As mentioned in the previous PR, this example is not working (no highlighted lines).

Reply:
Yes, it's a bug in the LineHighlight plugin. If you trigger a repaint (e.g. slightly resize the window) it will kick it. It's super weird, still debugging it.

Reply:
After a lot of head-scratching (while trying our different Ember events combined with the only two hooks available in Prism) the only hack I could find was to [trigger a delayed resize](https://github.com/hashicorp/design-system/pull/1687/commits/1fc56a545fa14869cec8ee679116a9082a73cc03#diff-54fbcfe4d8d03db33af48acaa4408fe0b24c37526daf2556b17bef1d5a0cdfa9R143-R147). The fix is not available in this branch, but if you'd like to see it in action you can try [navigating to the CodeBlock page in the showcase](https://hds-showcase-git-hds-2595-codeblock-component-hashicorp.vercel.app/). We can do a final check after we merge the docs branch in the component, but I tested the 'fix' locally and it works on the website too.